include/geos/geom/FixedSizeCoordinateSequence.h

```diff
@@ -37,7 +37,11 @@ namespace geom {
         std::unique_ptr<CoordinateSequence> clone() const final override {
             auto seq = detail::make_unique<FixedSizeCoordinateSequence<N>>(dimension);
             seq->m_data = m_data;
+#if defined __GNUC__ && ((__GNUC__ == 4) && (__GNUC_MINOR__ == 8))
```

Comment:
This is pretty gross. I thought the remove was in fact required by the C++11 standard (and not required in C++14?)

Reply:
Yeah, looking at the build failures it seems that old compilers can't auto cast between the declared and the returned type. I remember I did some trick in my previous PR to handle a similar situation, so I'll try to go that route instead.

Reply:
Here's the reference I was looking for: http://www.open-std.org/jtc1/sc22/wg21/docs/cwg_defects.html#1579

If you're strictly following C++11, it's the older compilers that are correct, and newer compilers that are applying a non-standard extension. I'm surprised that clang generates an error with `-std=c++11`.

Reply:
I don't see any `-std=c++11` in the autotools build. Maybe that's the way to go as long as the project is based on c++11.

Reply:
Seems like the main issue is that I don't get --std=c++11 with clang with the current autotools macro, but I get it with the one we have for Postgis, so I'll open a new PR to start with that and build only tackle whatever warnings are left after that.